src/fit.jl

```diff
@@ -152,7 +169,9 @@ function eval_metrics!(results::Dict{String,Dict{String,Vector{Float64}}},
                     best_score[metric_idx, test_idx] = maximize_score
                     best_iter[metric_idx, test_idx] = iter
                 elseif iter - best_iter[metric_idx, test_idx] >= estimator.early_stopping_round
-                    shrinkresults!(results, best_iter[metric_idx, test_idx])
+                    # need to fix this, its clearly broken in light of latest updates
+                    # This will shrink it up to the current stored metric
+                    shrinkresults!(results, metrics_store_idx)
```

Comment:
I can't exactly put my finger on why but this is incorrect (in the sense that the previous logic was wanting to shrink values to best iteration and here we just shrink if we're about to stop to current storage index)

Reply:
I think this is ok -- it might have always been wrong with metrics_freq > 1